### PR TITLE
86cueuap5/UI/SignIn-SignUp

### PR DESCRIPTION
**Pull request by nijin-26**
### Changes Made
- Implemented Auth Modal with login and signup UI.
- Integrated Formik with Yup for form validation.

### CODE REVIEW

File: src/components/ui/AuthModal/AuthModal.styled.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import { Box, Tab } from '@mui/material';
+
+export const AuthModalWrapper = styled(Box)(({ theme }) => ({
+    position: 'absolute',
+    top: '50%',
+    left: '50%',
+    transform: 'translate(-50%, -50%)',
+    width: '60rem',
+    minWidth: '45rem',
+    // padding: '2rem',
```
Comment (RJR-003) [inline src/components/ui/AuthModal/AuthModal.styled.ts]: remove unnecessary comments

File: src/components/ui/AuthModal/SignIn/SignIn.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import getValidationSchema from './validationSchema';
+
+type TSignInProps = {
+    closeModal: () => void;
+};
+
+const SignIn = ({ closeModal }: TSignInProps) => {
+    const { t } = useTranslation('auth');
+
+    const initialValues = {
```
Comment (RJR-003) [inline src/components/ui/AuthModal/SignIn/SignIn.tsx]: You can move these outside the component

File: src/components/ui/AuthModal/SignUp/SignUp.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+import getValidationSchema from './validationSchema';
+
+type TSignUpProps = {
+    closeModal: () => void;
+};
+
+const SignUp = ({ closeModal }: TSignUpProps) => {
+    const { t } = useTranslation('auth');
+
+    const initialValues = {
```
Comment (RJR-003) [inline src/components/ui/AuthModal/SignUp/SignUp.tsx]: Move these outside the component

**Review by RJR-003 — CHANGES_REQUESTED** [review]

File: src/components/ui/AuthModal/SignIn/SignIn.tsx (hunk 2 above)
Comment (nijin-26) [reply src/components/ui/AuthModal/SignIn/SignIn.tsx]: Updated in the recent commit.

File: src/layout/Header/Header.tsx [hunk 4]
```diff
@@ -19,38 +19,47 @@ import testProfile from '../../assets/person1.jpeg';
...
-const settings = ['My bookings', 'Logout'];
+import { AuthModal } from '../../components';
 
 export const Header = () => {
     const { t } = useTranslation('headerFooter'); // mention "ns2" to include values from ns2.json
     const themeMode = useAppSelector((state) => state.theme.currentTheme);
     const dispatch = useAppDispatch();
 
     const [isLoginClicked, setLoginClick] = useState(false);
-    const [anchorElUser, setAnchorElUser] = useState<null | HTMLElement>(null);
```
Comment (sania-salim) [inline src/layout/Header/Header.tsx:31]: Could use more specific names for anchorEl states

File: src/components/ui/AuthModal/SignIn/SignIn.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+const initialValues = {
+    email: '',
+    password: '',
+};
+
+const SignIn = ({ closeModal }: TSignInProps) => {
+    const { t } = useTranslation('auth');
+
+    const handleSubmit = () => {
+        console.log('Sign in submit');
```
Comment (sania-salim) [inline src/components/ui/AuthModal/SignIn/SignIn.tsx]: Could remove logs to console when submit functionality is implemented.

File: src/components/ui/AuthModal/SignUp/validationSchema.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+        email: Yup.string()
+            .email(t('invalidEmailMsg'))
+            .required(t('emailRequired')),
+
+        phone: Yup.string()
+            .matches(/^[6-9]\d{9}$/, t('invalidPhone'))
+            .required(t('phoneRequired')),
+
+        password: Yup.string()
+            .min(6, t('invalidPhone'))
```
Comment (abhijith-biju) [inline src/components/ui/AuthModal/SignUp/validationSchema.ts]: I think you have displayed the wrong error message.

File: src/components/ui/AuthModal/SignIn/validationSchema.ts
Comment (abhijith-biju) [inline src/components/ui/AuthModal/SignIn/validationSchema.ts:1]: I think required and min-length validations can be bypassed by entering blank characters.

**Review by abhijith-biju — CHANGES_REQUESTED** [review]

File: src/components/ui/AuthModal/SignIn/SignIn.tsx (hunk 5 above)
Comment (nijin-26) [reply src/components/ui/AuthModal/SignIn/SignIn.tsx]: I have updated it in the recent commit.

File: src/components/ui/AuthModal/SignUp/validationSchema.ts (hunk 6 above)
Comment (nijin-26) [reply src/components/ui/AuthModal/SignUp/validationSchema.ts]: Fixed it.

**Review by abhijith-biju — APPROVED** [review]

**Review by sania-salim — APPROVED** [review]

**Review by RJR-003 — APPROVED** [review]

File: src/components/ui/AuthModal/SignUp/validationSchema.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        fullName: Yup.string()
+            .trim()
+            .min(3, t('minNameMsg'))
+            .max(50, t('maxNameMsg'))
+            .required(t('nameRequired')),
+
+        email: Yup.string()
+            .trim()
+            .email(t('invalidEmailMsg'))
+            .required(t('emailRequired')),
```
Comment (lekshmi615213) [inline src/components/ui/AuthModal/SignUp/validationSchema.ts:18]: email format validation ?

**Review by lekshmi615213 — APPROVED** [review]

File: src/components/ui/AuthModal/SignUp/validationSchema.ts (hunk 7 above)
Comment (nijin-26) [reply src/components/ui/AuthModal/SignUp/validationSchema.ts:18]: I have resolved it in the recent commit.

**Pull request merged**